src/librbd/cache/pwl/Utils.h

```diff
@@ -0,0 +1,256 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
```

Comment:
Nit: the vast majority of these are types -- it's just the few that were in the `Types.cc` file that were actually utility functions.

Reply:
I just found a comment is not needed, and other functions are for WriteLogPmemEntry and WriteLogPoolRoot.

Reply:
As such l_librbd_pwl_XYZ enums and constants are not related to encoding/decoding, I moved them out of `Types.*`. So that `Types.*` just includes the data structures related to encoding/decoding.

Reply:
IMHO it really belong in `Types.h` since they are types and they are certainly not utilities.

Reply:
OK. I moved l_librbd_pwl_XYZ enums and constants back to Types.h.

Reply:
Honestly, the majority of this file is still types. All this effort just doesn't seem worth it to me to avoid pulling in `Types.cc` into `librbd_types_srcs` for the RWL-only case.

Reply:
That will be better if we include Types.cc into librbd_types_srcs if RWL or SSD is enabled. I revert the change. How is it now?